pkgs/applications/misc/octoprint/default.nix

```diff
@@ -209,6 +215,15 @@ let
                 rm pytest.ini
               '';
 
+              udevRules = ''
+                # Change /dev/vchiq from root:root 0600 to root:dialout  0660 for vcgencmd to work
+                SUBSYSTEM=="vchiq", GROUP="dialout", MODE="0660"
```

Comment:
The commit adding dialout to SupplementaryGroups https://github.com/NixOS/nixpkgs/commit/f8df7606683fc9fbce68c8c363d1318dab2ed199 says dailout is for serial devices. I do lack the domain knowledge about this, what is vchiq?

Reply:
https://raspberrypi.stackexchange.com/a/54590
The post suggests that vchiq is a decently dangerous interface to expose to random programs, hence the somewhat restrictive permissions by default. Not sure if we really want to loosen that permission.

Reply:
It is a raspberry pi specific program to check internals about the hardware (like throttling). On native devices running something like ubuntu etc it seems to be owned by the `video` group with Mode 0660.
I decided against changing the group to `video` and adding `octoprint` to that group so to give the least privileges to the service, while still maintaining the function to run it.

Reply:
Well it seems some people set it to Mode `777` for programs to `just work`.
I think `dialout` is a specific enough group to get access to that device. But I'm open to suggestions to change that

Reply:
Can we use ACL to grant only `octoprint` access to vchiq. `tmpfiles.d` can be used to set ACL on files.